src/rgw/driver/rados/rgw_tools.h

```diff
@@ -62,10 +62,13 @@ inline int rgw_shards_max()
 // only called by rgw_shard_id and rgw_bucket_shard_index
 static inline int rgw_shards_mod(unsigned hval, int max_shards)
 {
-  if (max_shards <= RGW_SHARDS_PRIME_0) {
+  if (max_shards <= 0) {
+    return -1;
+  } else if (max_shards <= RGW_SHARDS_PRIME_0) {
     return hval % RGW_SHARDS_PRIME_0 % max_shards;
+  } else {
+    return hval % RGW_SHARDS_PRIME_1 % max_shards;
```

Comment:
Just to be sure I follow, the goal to clamp them to [1, RGW_SHARDS_PRIME_0), that itself within [1, max_shards)?

Reply:
It's more than just clamping -- we want to distribute the values evenly across the shards, so we use modulo. Clamping, if I understand it, would map all values >= max_shards to (max_shards - 1), which would not distribute the values evenly across the shards.

Reply:
Aaah, you're absolutely right! Thanks for clarifying.